### PR TITLE
Update OnEvaluation event to have FSharpSymbolUse information available

### PR DESCRIPTION
**Pull request by 7sharp9**
This also simplifies the API as FSharpSymbolUse contains the display
context and range.  This allows for a better accuracy by using pattern
matching on the symbols.

### CODE REVIEW

**Comment by 7sharp9** [comment]
Ok, added simple test.

**Comment by 7sharp9** [comment]
@dsyme ^

**Comment by 7sharp9** [comment]
Ping @anyone, is this ok or?   

I would like to get this merged so I don't have to use a branch , and instead use the nuget :-)

File: src/fsharp/fsi/fsi.fs [hunk 1]
```diff
@@ -1120,23 +1119,23 @@ type internal FsiDynamicCompiler
 
         let filteredByVal =
             allNames
-            |> List.choose (function
```
Comment (dsyme) [inline src/fsharp/fsi/fsi.fs]: THis code is still kind of strange (as it was in the originakl version). It reports the top-level names defined, e.g. "x" in "let x = 1", but doesn't report any modules or types defined.

That's not related to this PR but the feature should eventually be adjusted to do a more systematic job on reporting freshly defined items.

Comment (7sharp9) [inline src/fsharp/fsi/fsi.fs]: Sure, the filter could be removed and to return non val items.  As far as I remember all the global top levels are here too like bigint etc

**Pull request merged**